src/utils.js

```diff
@@ -33,7 +33,7 @@ export const getMyTxBalance = (tx, myKeys) => {
       continue;
     }
     if (txout.decoded && txout.decoded.address
```

Comment:
Shouldn't this whole method (`getMyTxBalance()`) be replaced by the wallet object?

Reply:
We have another method `getBalance` in `utils.js` that should be full removed and I haven't noticed. I will look better on how we calculate the balance here and try to use only the wallet object.

Reply:
I've removed the method `getBalance` but I am currently keeping this `getMyTxBalance`. This method is used when receiving a new tx to get some information about the tokens and balance of the new tx for the wallet. We use this to update some internal redux variables, including the tokens list.

What do you think? It's not exactly getting the wallet balance, I feel we could keep like this.

Reply:
I feel the HathorWallet instance should handle all of this. When a new tx is received, the Hathor Wallet instance will update its internals including the list of tokens. What do you think?

Reply:
I agree with you and the HathorWallet instance already has the list of tokens. The wallet instance has a method `getBalance`, which receives a token and returns the balance of it. I will explain better what this method (`getMyTxBalance`) is doing:

Because some calculus in the lib are expensive before we integration with wallet service, we pre save some information on redux. When the app receives a new tx in the ws we execute the following actions:

- Call `getMyTxBalance` to get know which tokens are spent/received for the wallet in this tx (we might have an output for HTR but it's not for this wallet and this wallet is only receiving PedroCoin).
- With this information we (i) update the PedroCoin history list, and (ii) call `wallet.getBalance` for PedroCoin token to update the token balance.

Then the method `getMyTxBalance` is used to know which tokens are spent/received in the new tx, in order to run the updates. We will always need to update redux variables and I think this is the best approach, I don't feel we are running any lib internal operation here.

This will change with the wallet service integration but I think we should keep like this for now.

Reply:
I think that the wallet must be just a UI/UX over the `HathorWallet` instance. It doesn't have to do any processing except update the UI when needed. Maybe we should move `getMyTxBalance` to the wallet-lib, so the UI can efficiently update itself without recalculating all the balances. Does it make sense?

Reply:
Yeah, it makes sense. I am now using a method `getTxBalance` from the wallet instance.